src/gpu/kmeans/KmMatrix/utils.cuh

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright 2018 H2O.ai, Inc.
+ * License   Apache License Version 2.0 (see LICENSE for details)
+ */
+
+#ifndef UTILS_CUH_
+#define UTILS_CUH_
+
+#include "GpuInfo.cuh"
+
+namespace H2O4GPU {
+namespace KMeans {
+
+M_DEVINLINE size_t global_thread_idx () {
+  return threadIdx.x + blockIdx.x * blockDim.x;
+}
+
+M_DEVINLINE size_t global_thread_idy () {
+  return threadIdx.y + blockIdx.y * blockDim.y; 
+}
+
+M_DEVINLINE size_t grid_stride_x () {
+  return blockDim.x * gridDim.x;
+}
+
+M_DEVINLINE size_t grid_stride_y () {
+  return blockDim.y * gridDim.y;
+}
+
+template <typename T1, typename T2>
+T1 M_HOSTDEVINLINE div_roundup(const T1 a, const T2 b) {
+  return static_cast<T1>(ceil(static_cast<double>(a) / b));
+}
+
+
+// Work around for shared memory
+// https://stackoverflow.com/questions/20497209/getting-cuda-error-declaration-is-incompatible-with-previous-variable-name
```

Comment:
This should be named KmSharedMem instead of KmShardMem as the word "shard" can have a different meaning in the context of GPU programming. Although we do not seem have an explicit code style for this project, Google style recommends not using abbreviated words in variable names. So this would then be: KmSharedMemory.

As above I also don't think this code is specific to kmeans so should go in a shared utilities file.

Reply:
Will be changed.